tests/scripts/test_scripts.py

```diff
@@ -301,65 +282,87 @@ def test_train_dagger_warmstart(tmpdir):
-            ),
             agent_path=policy_path,
         ),
     )
     assert run_warmstart.status == "COMPLETED"
     assert isinstance(run_warmstart.result, dict)
 
 
-def test_train_dagger_error_and_exceptions(tmpdir):
-    with pytest.raises(Exception, match=".*expert_policy_path cannot be None.*"):
+def test_train_bc_main_with_none_demonstrations_raises_value_error(tmpdir):
+    with pytest.raises(ValueError, match=".*n_expert_demos.*rollout_path.*"):
         train_imitation.train_imitation_ex.run(
-            command_name="dagger",
-            named_configs=["cartpole"] + ALGO_FAST_CONFIGS["imitation"],
+            command_name="bc",
+            named_configs=["seals_cartpole"] + ALGO_FAST_CONFIGS["imitation"],
             config_updates=dict(
                 common=dict(log_root=tmpdir),
-                demonstrations=dict(rollout_path=CARTPOLE_TEST_ROLLOUT_PATH),
-                dagger=dict(
-                    expert_policy_type="ppo",
-                    expert_policy_path=None,
-                ),
+                demonstrations=dict(n_expert_demos=None),
             ),
         )
 
 
-def test_train_bc_main(tmpdir):
-    run = train_imitation.train_imitation_ex.run(
+@pytest.fixture(
+    params=[
+        "expert_from_path",
+        "expert_from_huggingface",
+        "random_expert",
+        "zero_expert",
+    ],
+)
+def bc_config(tmpdir, request):
+    expert_config = dict(
+        expert_from_path=dict(
+            policy_type="ppo",
+            loader_kwargs=dict(path=CARTPOLE_TEST_POLICY_PATH / "model.zip"),
+        ),
+        expert_from_huggingface=dict(
+            policy_type="ppo-huggingface",
+            loader_kwargs=dict(env_id="seals/CartPole-v0"),
+        ),
+        random_expert=dict(policy_type="random"),
+        zero_expert=dict(policy_type="zero"),
+    )[request.param]
+    return dict(
         command_name="bc",
-        named_configs=["cartpole"] + ALGO_FAST_CONFIGS["imitation"],
+        named_configs=["seals_cartpole"] + ALGO_FAST_CONFIGS["imitation"],
         config_updates=dict(
             common=dict(log_root=tmpdir),
+            expert=expert_config,
             demonstrations=dict(rollout_path=CARTPOLE_TEST_ROLLOUT_PATH),
         ),
     )
+
+
+def test_train_bc_main(bc_config):
+    run = train_imitation.train_imitation_ex.run(**bc_config)
     assert run.status == "COMPLETED"
     assert isinstance(run.result, dict)
 
 
 def test_train_bc_warmstart(tmpdir):
     run = train_imitation.train_imitation_ex.run(
         command_name="bc",
-        named_configs=["cartpole"] + ALGO_FAST_CONFIGS["imitation"],
+        named_configs=["seals_cartpole"] + ALGO_FAST_CONFIGS["imitation"],
         config_updates=dict(
             common=dict(log_root=tmpdir),
             demonstrations=dict(rollout_path=CARTPOLE_TEST_ROLLOUT_PATH),
+            expert=dict(
+                policy_type="ppo-huggingface",
+                loader_kwargs=dict(env_id="seals/CartPole-v0"),
```

Comment:
How does `env_id` differ from the `env_name` automatically set in `expert.get_expert_policy`?

Reply:
Good catch. This is a relic from when the `env_name` was not automatically set in the expert ingredient.
That is one of the drawbacks of not modelling the different loaders as sacred ingredients: we don't catch unused configuration parameters. I fixed it.